lib/instrumentation/index.js

```diff
@@ -358,35 +361,43 @@ Instrumentation.prototype._startHook = function () {
 
   this._agent.logger.debug('adding hooks to Node.js module loader');
 
-  this._ritmHook = new RitmHook(this._patches.keys, function (
-    exports,
-    name,
-    basedir,
-  ) {
-    const enabled = self._isModuleEnabled(name);
-    var pkg, version;
-
-    const isHandlingLambda =
-      self._lambdaHandlerInfo && self._lambdaHandlerInfo.module === name;
-
-    if (!isHandlingLambda && basedir) {
-      pkg = path.join(basedir, 'package.json');
-      try {
-        version = JSON.parse(fs.readFileSync(pkg)).version;
-      } catch (e) {
-        self._agent.logger.debug(
-          'could not shim %s module: %s',
-          name,
-          e.message,
-        );
-        return exports;
+  this._ritmHook = new RitmHook(
+    this._patches.keys,
+    // XXX: found this was not enabled so our redis instrumentations
+    // may not be working as for now
+    { internals: true },
+    function (exports, name, basedir) {
+      // RITM returns the path with extension for sub-modules
+      // so we need to make sure is not there when patching
+      if (name.endsWith('.js')) {
+        name = name.substring(0, name.length - 3);
       }
```

Comment:
This was only needed when `internals: true` was being used?

Reply:
That's correct. I'll check again how the hook works with using your diff and remove the RITM options

Reply:
With the latest commit this is not necessary. The hook delegates the extension handling to the patch function.